mlflow/utils/environment.py

```diff
@@ -439,23 +407,46 @@ def infer_pip_requirements(model_uri, flavor, fallback=None):
         flavor: The flavor name of the model.
         fallback: If provided, an unexpected error during the inference procedure is swallowed
             and the value of ``fallback`` is returned. Otherwise, the error is raised.
+        timeout: If specified, the inference operation is bound by the timeout (in seconds).
 
     Returns:
         A list of inferred pip requirements (e.g. ``["scikit-learn==0.24.2", ...]``).
 
     """
+    raise_on_error = MLFLOW_REQUIREMENTS_INFERENCE_RAISE_ERRORS.get()
+
+    if timeout and is_windows():
+        timeout = None
+        _logger.warning(
+            "On Windows, timeout is not supported for model requirement inference. Therefore, "
+            "the operation is not bound by a timeout and may hang indefinitely. If it hangs, "
+            "please consider specifying the signature manually."
+        )
+
     try:
```

Comment:
note: This was the try-catch that suppressed the exception

Reply:
Great find